server/insights/src/main/java/org/infinispan/server/insights/scheduler/InfinispanInsightsScheduler.java

```diff
@@ -0,0 +1,89 @@
+package org.infinispan.server.insights.scheduler;
+
+import static com.redhat.insights.InsightsErrorCode.ERROR_SCHEDULED_SENT;
+
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.util.concurrent.BlockingManager;
+
+import com.redhat.insights.InsightsException;
+import com.redhat.insights.InsightsScheduler;
+import com.redhat.insights.config.InsightsConfiguration;
+import com.redhat.insights.logging.InsightsLogger;
+
+/**
+ * Copy and adapted from {@link com.redhat.insights.InsightsCustomScheduledExecutor}.
+ * Differently from the original scheduler,
+ * instead of creating a new {@link java.util.concurrent.ScheduledExecutorService},
+ * it delegates to {@link BlockingManager}.
+ */
+public class InfinispanInsightsScheduler implements InsightsScheduler {
+
+   private final InsightsLogger logger;
+   private final InsightsConfiguration configuration;
+   private final BlockingManager blockingManager;
+
+   public InfinispanInsightsScheduler(InsightsLogger logger, InsightsConfiguration configuration,
+                                      BlockingManager blockingManager) {
+      this.logger = logger;
+      this.configuration = configuration;
+      this.blockingManager = blockingManager;
+   }
+
+   @Override
+   public ScheduledFuture<?> scheduleConnect(Runnable sendConnect) {
+      return scheduleAtFixedRate(sendConnect, 0, configuration.getConnectPeriod().getSeconds(), TimeUnit.SECONDS);
+   }
+
+   @Override
+   public ScheduledFuture<?> scheduleJarUpdate(Runnable sendNewJarsIfAny) {
+      return scheduleAtFixedRate(sendNewJarsIfAny,
+            configuration.getUpdatePeriod().getSeconds(),
+            configuration.getUpdatePeriod().getSeconds(),
+            TimeUnit.SECONDS);
+   }
+
+   @Override
+   public boolean isShutdown() {
+      return false;
+   }
+
+   @Override
+   public void shutdown() {
+      // do nothing, since the resource is shared by different Infinispan module
+   }
+
+   @Override
+   public List<Runnable> shutdownNow() {
+      return null;
+   }
+
+   public ScheduledFuture<?> scheduleAtFixedRate(Runnable command, long initialDelay, long period, TimeUnit unit) {
+      Runnable wrapped =
+            () -> {
+               try {
+                  command.run();
+               } catch (InsightsException ix) {
+                  logger.error(
+                        ERROR_SCHEDULED_SENT.formatMessage(
+                              "Red Hat Insights client scheduler shutdown, scheduled send failed: "
+                                    + ix.getMessage()),
+                        ix);
+                  shutdown();
```

Comment:
Shutdown does nothing.. do we want to have some type of running state for the InfinispanInsightsScheduler  so we can actually stop the ScheduledFuture  as well?

Reply:
You definitely right! I agree, we need to stop the ScheduledFuture without stopping the scheduler executor.